modules/annotations/shared/src/main/scala/io/circe/derivation/annotations/JsonCodec.scala

```diff
@@ -87,11 +87,11 @@ private[derivation] final class GenericJsonCodecMacros(val c: blackbox.Context)
 
   private[this] def codecFrom(tree: Tree): JsonCodecType =
     tree.tpe.dealias match {
-      case t if t <:< typeOf[Configuration.Codec] =>
+      case t if t == typeOf[Configuration.Codec] =>
```

Comment:
`equals` is not guaranteed to work correctly on scala-reflect types. Maybe `=:=`?

Reply:
Good catch, I hadn't noticed these changes. @aparo, is there a reason to change these? I fixed this a few weeks ago because I was getting inaccurate warnings in some contexts.